frontend/src/app/components/mempool-graph/mempool-graph.component.ts

```diff
@@ -100,12 +99,136 @@ export class MempoolGraphComponent implements OnInit, OnChanges {
     return finalArray;
   }
 
+  filterLabelDates(labels: any) {
+    let labelDates = [];
+
+    if (labels.length > 0) {
+      for (let index = 0; index < labels.length; index++) {
+        const labelDate = new Date(labels[index]);
+
+        // add the first date to the array
+        if (index === 0) {
+          labelDates.push(labelDate.toISOString());
+        } else {
+
+          // mobile sizes
+          if(window.innerWidth < 500) {
+
+            if (this.windowPreference === '2h') {
+              if (labelDate.getMinutes() % 60 === 0) {
+                labelDates.push(labelDate.toISOString());
+              }
+            }
+
+            if (this.windowPreference === '24h') {
+              if (labelDate.getMinutes() % 60 === 0 &&
+                  labelDate.getHours() % 2 === 0) {
+                labelDates.push(labelDate.toISOString());
+              }
+            }
+
+            if (this.windowPreference === '1w') {
+              if (labelDate.getMinutes() % 60 === 0 &&
+                  labelDate.getHours() % 6 === 0) {
+                labelDates.push(labelDate.toISOString());
+              }
+            }
+
+            if (this.windowPreference === '1m') {
+              if (labelDate.getMinutes() % 60 === 0 &&
+                  labelDate.getHours() % 12 === 0) {
```

Comment:
It looks like you should be targeting midnight instead of noon here
<img width="1221" alt="Screen Shot 2021-11-21 at 1 29 24" src="https://user-images.githubusercontent.com/232186/142733902-91df3fea-ef09-4123-be0d-a59bf5bee741.png">